pkg/git/git.go

```diff
@@ -65,7 +65,7 @@ func Fetch(logger *zap.SugaredLogger, spec FetchSpec) error {
 	}
 
 	if spec.Revision == "" {
-		spec.Revision = "master"
+		spec.Revision = "HEAD"
```

Comment:
I think we should run `git symbolic-ref refs/remotes/origin/HEAD` only if `spec.Revision` is empty and extract the default branch name from it. So something like :

```go
remoteHead, err := run(logger, "", "symbolic-ref", "refs/remotes/origin/HEAD")
if err != nil {
	return err
}
spec.Revision = strings.ReplaceAll(remoteHead, "refs/remotes/origin/", "") // this would display `master` or `main`, or …
```

Reply:
@vdemeester : I tried this out locally on my terminal and also as you suggested but this didn't worked out. On my terminal I got the following error:-
```
[vinjain@vinjain output]$ git remote show origin
* remote origin
  Fetch URL: https://github.com/spring-projects/spring-petclinic.git
  Push  URL: https://github.com/spring-projects/spring-petclinic.git
  HEAD branch: main
  Remote branches:
    gh-pages  new (next fetch will store in remotes/origin)
    main      new (next fetch will store in remotes/origin)
    wavefront new (next fetch will store in remotes/origin)
[vinjain@vinjain output]$ git symbolic-ref refs/remotes/origin/HEAD
fatal: ref refs/remotes/origin/HEAD is not a symbolic ref
```

Reply:
@vinamra28 you are right, it needs to happend after the fetch :thinking: